Select_Flick_Week2/FavoriteMovieService.swift

```diff
@@ -10,23 +10,59 @@ import Foundation
 import RealmSwift
 
 class FavoriteMovieService {
+
+    static func getFavoriteMovies(username: String) -> UserFavoriteMoviesDTO {
+        let realm = try! Realm()
+        let predicate = NSPredicate(format: "username = %@", username)
+        let favoriteMovies = realm.objects(UserFavoriteMoviesDTO.self).filter(predicate)
+        if favoriteMovies.count == 0 {
+            return UserFavoriteMoviesDTO()
+        } else {
+            return (favoriteMovies.first)!
+        }
+    }
     
     static func clickFavoriteButton(username: String, movieId: Int){
         let realm = try! Realm()
         let predicate = NSPredicate(format: "username = %@", username)
-        let favoriteMovies = realm.objects(FavoriteMovieDTO.self).filter(predicate)
+        let favoriteMovies = realm.objects(UserFavoriteMoviesDTO.self).filter(predicate)
+        realm.beginWrite()
         if favoriteMovies.count == 0 {
             // add new
+            var userFavMovies = UserFavoriteMoviesDTO()
+            userFavMovies.username = username
+            var movie = FavoriteMovieDTO()
+            movie.movieId = movieId
+            movie.isLike = true
+            movie.transactionDate = NSDate()
+            userFavMovies.favouriteMovies.append(movie)
+            //try! realm.commitWrite() {
+                realm.create(UserFavoriteMoviesDTO.self, value: userFavMovies)
+            //}
+            try! realm.commitWrite()
+            
         } else {
-            var favoriteMovies = favoriteMovies.first?.favouriteMovies
-            // update
-            if favoriteMovies!.indexForKey(movieId) != nil {
-                favoriteMovies![movieId] = !(favoriteMovies![movieId]!)
-            } else {
-
+            var isFound:Bool = false
+            for movie in (favoriteMovies.first?.favouriteMovies)! {
```

Comment:
I think this logic is a little hard to follow because `isFound` is a confusing terminology. Maybe something is `isFavorited`?